src/Hint/Base.hs

```diff
@@ -179,7 +179,8 @@ showGHC :: (MonadInterpreter m, GHC.Outputable a) => a -> m String
 showGHC a
  = do unqual <- runGhc GHC.getPrintUnqual
       withDynFlags $ \df ->
-        return $ GHC.showSDocForUser df unqual (GHC.ppr a)
+        -- TODO: get unit state from somewhere?
```

Comment:
everything we evaluate within a `runInterpreter` block is evaluated inside a single compilation unit, whose name is randomly generated at the beginning of the block. it does make sense to associate the compilation unit's state to the `InterpreterT` monad.

Reply:
Makes sense; however, the tests do pass. So maybe merge this now to just get GHC 9.2 support, and add a ticket to add unit state?

Unless you have a quick way to resolve this now

Reply:
[done](https://github.com/haskell-hint/hint/issues/140), thanks for the suggestion! since this is only a `Show` function, I am confident that nothing will go horribly wrong (at worst things will go slightly wrong) if we leave it like this.